### PR TITLE
OpenFileManager for opening with the native file manager and optional file selection support

### LINKED ISSUE #3197: Is there a folder and file runtime api?
1. open folder and select a specified file. Not a dialog,like windows `explorer.exe /select,absfilepath`. 
2. open a specified folder,not a dialog. 
3. open a specified file,not a dialog.

### PR DESCRIPTION
**Pull request by rcalixte**
Closes #3197

cc @Krzysztofz01

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change
  
Please select the option that is relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
  
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration using `wails doctor`.

- [ ] Windows
- [ ] macOS
- [ ] Linux
      
If you checked Linux, please specify the distro and version.
  
## Test Configuration

Please paste the output of `wails doctor`. If you are unable to run this command, please describe your environment in as much detail as possible.

# Checklist:

- [ ] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [x] My code follows the general coding style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Release Notes

- **New Features**
	- Introduced a new method to open the file manager with optional file selection highlighting.
	- Added support for Linux packaging and enhanced macOS builds.
	- New templates for SvelteKit have been introduced.

- **Bug Fixes**
	- Resolved issues with callbacks running on the main thread for macOS.
	- Fixed deadlocks, compilation issues, and improved cross-platform compatibility.

- **Documentation**
	- Updated changelog to reflect new features, fixes, and changes across versions.
	- Added documentation for events on the mkdocs website.

### CODE REVIEW

**Review by leaanthony — COMMENTED** [review]
Nice one! Do you think we could add some tests to ensure that the string processing does what we think it should?

**Comment by rcalixte** [comment]
> Nice one! Do you think we could add some tests to ensure that the string processing does what we think it should?

I've added some tests but it would help to have paths that are known at the time the tests are running.

**Comment by rcalixte** [comment]
> Nice one! Do you think we could add some tests to ensure that the string processing does what we think it should?

I think it's ready for review. Let me know if anything else should be changed.

File: v3/pkg/application/application.go [hunk 1]
```diff
@@ -1046,8 +1047,16 @@ func (a *App) Paths(selector Paths) []string {
 	return pathdirs[selector]
 }
 
+// OpenDirectory opens the system file explorer to the specified directory.
 func (a *App) OpenDirectory(path string) error {
```
Comment (Krzysztofz01) [inline v3/pkg/application/application.go:1051]: Do we want to leave both `OpenDirectory` and `OpenFileManager` methods in the runtime API?

Technically, `OpenDirectory` also opens the file manager but without the possibility to select a file

Comment (leaanthony) [reply v3/pkg/application/application.go:1051]: Yes, let's remove it!

**Review by leaanthony — APPROVED** [review]

**Pull request merged**